Formula/pyenv.rb

```diff
@@ -47,6 +49,20 @@ def install
   end
 
   test do
-    shell_output("eval \"$(#{bin}/pyenv init -)\" && pyenv versions")
+    # Create a fake python version and executable.
+    shell_output("mkdir -p \"$(pyenv root)/versions/1.2.3/bin\"")
+    shell_output("echo 'echo hello' > \"$(pyenv root)/versions/1.2.3/bin/foo\"")
+    shell_output("chmod +x \"$(pyenv root)/versions/1.2.3/bin/foo\"")
+
+    # Test versions.
+    versions = shell_output("eval \"$(#{bin}/pyenv init -)\" && pyenv versions").split("\n")
+    assert_equal 2, versions.length
+    assert_match(/\* system/, versions[0])
+    assert_equal("  1.2.3", versions[1])
+
+    # Test rehash.
+    shell_output("pyenv rehash")
```

Comment:
If you just want to check that this runs without errors:
```suggestion
    system "pyenv", "rehash"
```
`shell_output` is for when you want to capture stdout.